File: C_Programming/Practise_Programs/PP_145.c
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_145
//
//  Description       : This program checks whether the 3rd bit of
//                      a given number is ON or OFF using a bit mask.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 18/08/2026
//
//  Time Complexity   : O(1)
//  Space Complexity  : O(1)
//
//////////////////////////////////////////////////////////////////

#include<stdio.h>   
 
int main() 
{ 
    int iNo = 0; 
    int iMask = 4; 
    int iAns = 0; 
 
    printf("Enter Number : \n"); 
    scanf("%d", &iNo); 
 
    iAns = iNo & iMask; 
 
    if(iAns == iMask) 
    { 
        printf("3rd bit is ON\n"); 
    } 
    else 
    { 
        printf("3rd bit is OFF\n"); 
    } 
 
    return 0; 
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input :
//      12
//
//  Sample Output :
//
//      Enter Number :
//      3rd bit is ON
//
//////////////////////////////////////////////////////////////////

File: Java_Programming/Practise_Programs/PP_145.java
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_145
//
//  Description       : This program prints a left-aligned
//                      increasing star pattern using nested
//                      for loops. The number of stars increases
//                      by one in each successive row.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 02/09/2026
//
//  Time Complexity   : O(n^2)
//  Space Complexity  : O(1)
//
//////////////////////////////////////////////////////////////////

class PP_145
{
    public static void main(String A[])
    {
        int i, j = 0;

        for(i = 1; i <= 5; i++)
        {
            for(j = 1; j <= i; j++)
            {
                System.out.print(" *");
            }
            System.out.println();
        }
    }
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input :
//      No input required.
//
//  Sample Output :
//
//       *
//       * *
//       * * *
//       * * * *
//       * * * * *
//
//////////////////////////////////////////////////////////////////